### PR TITLE
fix(JS): Add callouts for Push Notifications and Social Sign-in from GH Issues

### LINKED ISSUE #5743: Create callout to warn about non-scoped attributes for OAuth/Social IDP's
**Describe the content issue**:
A clear and concise description of what the content issue is. Is there content missing or is there incorrect content?

See [comment](https://github.com/aws-amplify/amplify-js/issues/11623#issuecomment-1642117061) on Issue #11623 of JS repo. 

**URL page where content issue is**:

Create callout after initial set of paragraphs on [Social Sign-in](https://docs.amplify.aws/lib/auth/social/q/platform/js/#:~:text=OAuth%20support%20in%20Amplify%20uses%20Cognito%20User%20Pools%20and%20supports%20federation%20with%20social%20providers%2C%20which%20will%20automatically%20create%20a%20corresponding%20user%20in%20the%20User%20Pool%20after%20a%20login.%20OIDC%20tokens%20are%20available%20in%20the%20app%20after%20the%20application%20has%20completed%20this%20process.) page of Auth docs.

### PR DESCRIPTION
**Pull request by cwomack**
#### Description of changes:

Creates callouts that were determined to be necessary based on GitHub issues [#11623](https://github.com/aws-amplify/amplify-js/issues/11623#issuecomment-1642117061) (Social sign-in scopes) and [#11562](https://github.com/aws-amplify/amplify-js/issues/11562#issuecomment-1614882769) (Push Notifications request permissions).  Docs issues were created for each of these and this PR will address those along with satisfying the remaining tasks to close the associated issues within the JS repo.  

#### Related GitHub issue #, if available:
Resolves #5743, #5740

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [x] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [x] JS
- [x] iOS
- [x] Android
- [x] Flutter
- [x] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [ ] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://link.com)` 
            HTML: `<a href="https://link.com">link</a>`_

### When this PR is ready to merge, please check the box below
- [x] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/fragments/lib/auth/js/social.mdx [hunk 1]
```diff
@@ -4,6 +4,10 @@
 
 OAuth support in Amplify uses Cognito User Pools and supports federation with social providers, which will automatically create a corresponding user in the User Pool after a login. [OIDC](https://en.wikipedia.org/wiki/OpenID_Connect) tokens are available in the app after the application has completed this process.
 
+<Callout warning>
+ When configuring a Social sign-in through the Amplify CLI, it's important to exercise caution when designating attributes as "required." Different social identity providers have varied scopes in terms of the information they respond back to Cognito with.  User pool attributes that are initial set up as "required" cannot be changed later, and may require you to migrate the users or create a new user pool.
```
Comment (reesscot) [inline src/fragments/lib/auth/js/social.mdx:8]: grammar nit:
```suggestion
 When configuring Social sign-in through the Amplify CLI, it's important to exercise caution when designating attributes as "required." Different social identity providers have varied scopes in terms of the information they respond back to Cognito with.  User pool attributes that are initial set up as "required" cannot be changed later, and may require you to migrate the users or create a new user pool.  
```

Comment (cwomack) [reply src/fragments/lib/auth/js/social.mdx]: Nice catch Scott! Updated it in most recent commit where I added the callout across all platforms. Also lower-cased that "Social" at beginning.

**Review by abdallahshaban557 — APPROVED** [review]
LGTM

**Review by kvramyasri7 — APPROVED** [review]
LGTM @cwomack .

**Review by katiegoines — APPROVED** [review]

**Pull request merged**